tests/integration/test_docker.py

```diff
@@ -70,19 +69,34 @@ def test_docker():
 
 
 def test_read_file():
-    i = Image("fedora", tag="26")
+    i = DockerImage("fedora", tag="26")
     # i.pull()
-    c = Container(i)
+    c = DockerContainer(i)
     c.start("sleep infinity")
-    # we need to wait
-    time.sleep(1)
+    time.sleep(1)  # FIXME: replace by wait once available
     assert c.check_running()
     content = c.read_file("/etc/system-release")
     assert content == "Fedora release 26 (Twenty Six)\n"
     assert isinstance(content, str)
     assert_raises(subprocess.CalledProcessError, c.read_file, "/i/lost/my/banana")
 
 
+def test_http_client():
+    i = DockerImage("fedora", tag="26")
+    # i.pull()
+    c = DockerContainer(i)
+    c.start("python3 -m http.server --bind 0.0.0.0 8000")
+    time.sleep(1)  # FIXME: replace by wait once available
+    assert c.check_running()
+    r = c.http_request(port="8000")
+    assert "<!DOCTYPE HTML PUBLIC" in r.content
+    assert r.ok
+    r2 = c.http_request(path="/etc", port="8000")
+    assert "<!DOCTYPE HTML PUBLIC" in r2.content
+    assert "passwd" in r2.content
+    assert r2.ok
+
+
```

Comment:
This should be part of another PR and not related to Logging.

Reply:
This is part of another PR, see the description of this PR.